hit-test-transient-input.html

```diff
@@ -0,0 +1,257 @@
+      function onSelect(event) {
+        if (event.inputSource.handedness == "left" &&
+            reticleLeft.visible) {
+          // The reticle should already be positioned at the latest hit point, 
+          // so we can just use it's matrix to save an unnecessary call to 
+          // event.frame.getHitTestResults.
+          addARObjectAt(reticleLeft.matrix);
+        } else if (event.inputSource.handedness == "right" &&
+                   reticleRight.visible) {
+          addARObjectAt(reticleRight.matrix);
+        }
+      }
+
+      // Called every time a XRSession requests that a new frame be drawn.
+      function onXRFrame(t, frame) {
+        let session = frame.session;
+        let pose = frame.getViewerPose(xrRefSpace);
+
+        updateInputSources(session, frame, xrRefSpace);
+
+        reticleLeft.visible = false;
+        reticleRight.visible = false;
+
+        // If we have a hit test source, get its results for the frame
+        // and use the pose to display a reticle in the scene.
+        if (xrTransientInputHitTestSource && pose) {
+          let inputHitTestResults = frame.getHitTestResultsForTransientInput(xrTransientInputHitTestSource);
+          for (let inputHitTestResult of inputHitTestResults) {
+            if (inputHitTestResult.results.length > 0) {
+              let pose = inputHitTestResult.results[0].getPose(xrRefSpace);
+              if (inputHitTestResult.inputSource.handedness == "left") {
+                reticleLeft.visible = true;
+                reticleLeft.matrix = pose.transform.matrix;
+              } else {
+                reticleRight.visible = true;
+                reticleRight.matrix = pose.transform.matrix;
+              }
+            }
+          }
+        }
+
+        scene.startFrame();
+
+        session.requestAnimationFrame(onXRFrame);
+
+        scene.drawXRFrame(frame, pose);
+
+        scene.endFrame();
+      }
+
+      function updateInputSources(session, frame, refSpace) {
+        for (let inputSource of session.inputSources) {
+          let targetRayPose = frame.getPose(inputSource.targetRaySpace, refSpace);
+
+          // We may not get a pose back in cases where the input source has lost
+          // tracking or does not know where it is relative to the given frame
+          // of reference.
+          if (!targetRayPose) {
+            continue;
+          }
+
+          if (inputSource.targetRayMode == 'tracked-pointer') {
+            // If we have a pointer matrix and the pointer origin is the users
+            // hand (as opposed to their head or the screen) use it to render
+            // a ray coming out of the input device to indicate the pointer
+            // direction.
+            scene.inputRenderer.addLaserPointer(targetRayPose.transform);
+          }
+
+          if (!xrTransientInputHitTestSourceRequested) {
+            session.requestHitTestSourceForTransientInput({ profile: inputSource.profiles[0]})

```

Comment:
This isn't really in the spirit of how transient input hit tests are intended to be used. If you have an input source that you can get a profile from, you probably just want a regular hit test source, as it'll be easier to use and is likely to be cheaper on the implementation end.

A transient input source is typically thought of as one that may only be observable during a select event. `generic-touchscreen` is s good example here: We don't have a continuous input source that represents the touchscreen, but we may generate events that come from one. Since the select event may only persist for a single frame, and since `requestHitTestSource()` is asynchronous, the input source may disappear before the hit test source is fully created, and as a result you may never see hit test results from it.

Because `requestHitTestSourceForTransientInput()` takes a profile, you can do something like `session.requestHitTestSourceForTransientInput('generic-touchscreen')` to indicate to the system that you want to track hit tests for ANY source that includes the `generic-touchscreen` profile, even if it only persists for a frame. That way you can get the hit test results in the `select` events consistently.